frontend/rust-lib/flowy-document2/src/parser/external/parser.rs

```diff
@@ -0,0 +1,38 @@
+use crate::parser::external::utils::{flatten_element_to_block, parse_plaintext_to_nested_block};
+use crate::parser::parser_entities::{InputType, NestedBlock};
+use scraper::Html;
+
+/// External data to nested json parser.
+#[derive(Debug, Clone, Default)]
+pub struct ExternalDataToNestedJSONParser {
+  /// External data. for example: html string, plain text string.
+  external_data: String,
+  /// External data type. for example: [InputType]::Html, [InputType]::PlainText.
+  input_type: InputType,
+}
+
+impl ExternalDataToNestedJSONParser {
+  pub fn new(data: String, input_type: InputType) -> Self {
+    Self {
+      external_data: data,
+      input_type,
+    }
+  }
+
+  /// Format to nested block.
+  ///
+  /// Example:
+  /// ```json
```

Comment:
I don't get it. The JSON here seems vague

Reply:
I modify the comment
```
  /// Format to nested block.
  ///
  /// Example:
  /// - input html: <p><strong>Hello</strong></p><p> World!</p>
  /// - output json:
  /// { "type": "page", "data": {}, "children": [{ "type": "paragraph", "children": [], "data": { "delta": [{ "insert": "Hello", attributes: { "bold": true } }] } }, { "type": "paragraph", "children": [], "data": { "delta": [{ "insert": " World!", attributes: null }] } }] }
```